frontend/model/notifications/periodicNotifications.js

```diff
@@ -13,16 +12,13 @@ export const PERIODIC_NOTIFICATION_TYPE = {
   MIN30: '30MIN'
 }
 
-const every1MinTimeout = { notifications: [], state: {}, delay: MINS_MILLIS }
-const every5MinTimeout = { notifications: [], state: {}, delay: 5 * MINS_MILLIS }
-const every15MinTimeout = { notifications: [], state: {}, delay: 15 * MINS_MILLIS }
-const every30MinTimeout = { notifications: [], state: {}, delay: 30 * MINS_MILLIS }
+const ephemeralNotificationState: { notifications: any[], partition: Object, clearTimeout?: Function } = { notifications: [], partition: Object.create(null) }
 
```

Comment:
This looks like a great DRYing here:
- multiple `state: {}` are now merged into `partition: { stateKey: {} }`.
- multiple `notifications: []` are now merged into `ephemeralNotificationState.notifications: [...]`

Only a couple of concerns I see in this file below.

Reply:
Awesome